packages/components/src/Banner/Banner.tsx

```diff
@@ -53,22 +66,31 @@ export function Banner({
     );
   }
 
-  const flashClassNames = classnames(styles.flash, types[type]);
+  const flashClassNames = classnames(styles.flash, types[type], {
+    [styles.medium]: bannerWidth >= customWidths.medium,
+  });
   return (
     <>
       {showFlash && (
-        <div className={flashClassNames}>
-          <Text>{children}</Text>
-          {primaryAction && <Button {...primaryAction} />}
-
+        <div className={flashClassNames} ref={bannerRef} role="status">
+          <div className={styles.bannerContent}>
+            <Text>{children}</Text>
+            {primaryAction && (
+              <div className={styles.bannerAction}>
+                <Button {...primaryAction} />
+              </div>
+            )}
+          </div>
           {dismissible && (
-            <button
-              className={styles.closeButton}
-              onClick={handleClose}
-              aria-label="Close"
-            >
-              <Icon name="cross" color={iconColors[type]} />
-            </button>
+            <div className={styles.dismissWrapper}>
+              <button
+                className={styles.closeButton}
+                onClick={handleClose}
+                aria-label="Close"
+              >
+                <Icon name="cross" color={iconColors[type]} />
+              </button>
```

Comment:
We have a component `ButtonDismiss` that uses a `Button` component. Not sure if this is something we want to use here, but just double checking.

Reply:
I tried and it's a bit more opinionated (solid grey fill on hover/focus). I think we could make some changes to `ButtonDismiss`, but I think that could be a separate PR?